sources/packages/backend/apps/api/src/route-controllers/application-offering-change-request/application-offering-change-request.institutions.controller.ts

```diff
@@ -63,4 +68,92 @@ export class ApplicationOfferingChangeRequestInstitutionsController extends Base
       count: applications.count,
     };
   }
+
+  /**
+   * Gets all in progress application offering request changes.
+   * @param locationId location id.
+   * @param paginationOptions options to execute the pagination.
+   * @returns list of inprogress application offering request changes.
+   */
+  @HasLocationAccess("locationId")
+  @Get("in-progress")
+  async getInprogressApplications(
+    @Param("locationId", ParseIntPipe) locationId: number,
+    @Query() pagination: OfferingChangePaginationRequestedOptionsAPIInDTO,
+  ): Promise<
+    PaginatedResultsAPIOutDTO<InprogressApplicationOfferingChangesAPIOutDTO>
+  > {
+    const offeringChange =
+      await this.applicationOfferingChangeRequestService.getRequestedSummary(
+        locationId,
+        pagination,
+        [
+          ApplicationOfferingChangeRequestStatus.InProgressWithSABC,
+          ApplicationOfferingChangeRequestStatus.InProgressWithStudent,
+        ],
+      );
+    return {
```

Comment:
Just for better understanding, can we please refactor the below return something like this?

`const results = offeringChange.results.map((eachOfferingChange) => {
      const {
        application: {
          applicationNumber,
          id,
          student: { user },
          currentAssessment: { offering },
        },
        applicationOfferingChangeRequestStatus,
      } = eachOfferingChange;

      return {
        applicationNumber,
        applicationId: id,
        studyStartPeriod: offering.studyStartDate,
        studyEndPeriod: offering.studyEndDate,
        fullName: getUserFullName(user),
        status: applicationOfferingChangeRequestStatus,
      };
    });

    const count = offeringChange.count;

    return { results, count };`

Reply:
@guru-aot , is below complex to read `const {
application: {
applicationNumber,
id,
student: { user },
currentAssessment: { offering },
},
applicationOfferingChangeRequestStatus,
} = eachOfferingChange;`

Reply:
its just a suggestion @ann-aot, if you feel that is too complex. You can ignore the comment.

Reply:
In my opinion, it is more readable the way it currently is.